### PR TITLE
Some relatively reasonable error code modifications

### PR DESCRIPTION
**Pull request by zhhyu7**
## Testing
usrsocktest OK.

### CODE REVIEW

File: net/inet/inet_sockif.c [hunk 1]
```diff
@@ -407,7 +407,7 @@ static int inet_bind(FAR struct socket *psock,
   if (addrlen < minlen)
     {
       nerr("ERROR: Invalid address length: %d < %d\n", addrlen, minlen);
-      return -EBADF;
```
Comment (zhhyu7) [inline net/inet/inet_sockif.c:410]: https://github.com/linux-test-project/ltp/blob/3e0e60074495e1c350b634848e5dd1f7993ec2ef/testcases/kernel/syscalls/bind/bind01.c#L34-L35

File: net/socket/socketpair.c [hunk 2]
```diff
@@ -150,6 +150,11 @@ int socketpair(int domain, int type, int protocol, int sv[2])
       oflags |= O_CLOEXEC;
     }
 
+  if (type & SOCK_NONBLOCK)
```
Comment (zhhyu7) [inline net/socket/socketpair.c:153]: https://github.com/linux-test-project/ltp/blob/3e0e60074495e1c350b634848e5dd1f7993ec2ef/testcases/kernel/syscalls/socketpair/socketpair02.c#L37

File: net/inet/inet_sockif.c [hunk 3]
```diff
@@ -1809,23 +1809,15 @@ FAR const struct sock_intf_s *
...
-#ifdef NET_UDP_HAVE_STACK
-  if (type == SOCK_DGRAM && (protocol == 0 || protocol == IPPROTO_UDP))
-    {
-      return &g_inet_sockif;
-    }
-  else
-#endif
-#ifdef NET_TCP_HAVE_STACK
-  if (type == SOCK_STREAM && (protocol == 0 || protocol == IPPROTO_TCP))
+#if defined(NET_UDP_HAVE_STACK) || defined(NET_TCP_HAVE_STACK)
```
Comment (zhhyu7) [inline net/inet/inet_sockif.c:1812]: https://github.com/linux-test-project/ltp/blob/3e0e60074495e1c350b634848e5dd1f7993ec2ef/testcases/kernel/syscalls/socket/socket01.c#L34-L39

File: net/socket/socket.c [hunk 4]
```diff
@@ -82,6 +82,11 @@ int psock_socket(int domain, int type, int protocol,
   FAR const struct sock_intf_s *sockif = NULL;
   int ret;
 
+  if (type & ~(SOCK_CLOEXEC | SOCK_NONBLOCK | SOCK_TYPE_MASK))
```
Comment (zhhyu7) [inline net/socket/socket.c:85]: https://github.com/linux-test-project/ltp/blob/3e0e60074495e1c350b634848e5dd1f7993ec2ef/testcases/kernel/syscalls/socket/socket01.c#L32

File: net/socket/setsockopt.c [hunk 5]
```diff
@@ -82,7 +82,7 @@ static int psock_socketlevel_option(FAR struct socket *psock, int option,
 
   if (!value)
     {
-      return -EINVAL;
+      return -EFAULT;
```
Comment (zhhyu7) [inline net/socket/setsockopt.c:85]: https://github.com/linux-test-project/ltp/blob/3e0e60074495e1c350b634848e5dd1f7993ec2ef/testcases/kernel/syscalls/setsockopt/setsockopt01.c#L39-L40

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**